### PR TITLE
Player can join game son mobile

### PR DESCRIPTION
**Pull request by jonathan-marmelab**
This PR closes the following user story: https://trello.com/c/XRgyXl95/35-5-as-robert-i-want-to-play-even-when-philippe-is-not-nearby

Depends on #30 

Peek 10-03-2023 11-49

### CODE REVIEW

File: mobile/src/board/BoardPages.tsx [hunk 1]
```diff
@@ -1,9 +1,42 @@
...
 import BoardView from "./components/BoardView";
+import { useUserContext } from "../user/UserHooks";
+
+export function List() {
+  const [user, _] = useUserContext();
+  const [boards, setBoards] = useState<BoardListItem[]>([]);
+
+  useEffect(() => {
+    boardRepository.list(1).then(setBoards);
+  }, [user]);
```
Comment (djhi) [inline mobile/src/board/BoardPages.tsx:20]: You should use `react-query` for those API calls. Later though

**Review by djhi — CHANGES_REQUESTED** [review]
Now that you MUI, you should use their Typography List and Link components to get a nice and coherent UI

File: mobile/src/board/BoardPages.tsx [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
+
+  return (
+    <>
+      {!user && (
+        <strong>
+          You are not signed in! <br />
+          You can spectate these games:
+        </strong>
```
Comment (djhi) [inline mobile/src/board/BoardPages.tsx:28]: This could be a Typography

File: mobile/src/board/BoardPages.tsx [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+      )}
+      <ul>
+        {boards.map((board) => (
+          <li>
+            <Link to={`/board/${board.id}/view`}>Board #{board.id}</Link>
+          </li>
+        ))}
+      </ul>
```
Comment (djhi) [inline mobile/src/board/BoardPages.tsx:36]: Use List ?

File: mobile/src/board/BoardPages.tsx [hunk 4]
```diff
@@ -14,12 +47,37 @@ export function GetById() {
...
   if (board) {
+    if (board.remainingSeats > 0) {
+      return (
+        <>
+          <strong>Waiting for {board.remainingSeats} player(s)</strong>
+          <ul>
+            {board.players.map((player, i) => (
+              <li key={i}>{player?.name ?? "?"}</li>
+            ))}
+          </ul>
```
Comment (djhi) [inline mobile/src/board/BoardPages.tsx:65]: Same

File: mobile/src/board/components/BoardView.tsx [hunk 5]
```diff
@@ -57,13 +69,36 @@ const BoardView = ({
...
+              <div className="state__row__label">Your color</div>
+              <div className="state__row__value">{colorNames[user.color]}</div>
+            </div>
+            <div className="state__row">
+              <div className="state__row__label">Your target</div>
+              <div className="state__row__value">
+                {treasures[user.currentTarget]}
+              </div>
+            </div>
+          </div>
```
Comment (djhi) [inline mobile/src/board/components/BoardView.tsx:99]: Same

File: mobile/src/board/components/BoardView.tsx [hunk 6]
```diff
@@ -57,13 +69,36 @@ const BoardView = ({
...
       </div>
 
-      <TileView
-        boardTile={remainingTile}
-        canPlay={canPlay}
-        gameState={gameState}
-        onRotateRemainingTile={onRotateRemainingTile}
-        onInsertTile={onInsertTile}
-      />
+      <div className="state">
```
Comment (djhi) [inline mobile/src/board/components/BoardView.tsx:72]: use Stack

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by djhi — COMMENTED** [review]
Can you add a screenshot?

File: mobile/src/board/components/BoardView.tsx [hunk 7]
```diff
@@ -57,13 +72,57 @@ const BoardView = ({
...
+              gameState={gameState}
+              onRotateRemainingTile={onRotateRemainingTile}
+              onInsertTile={onInsertTile}
+              onMovePlayer={onMovePlayer}
+              playerTarget={user?.currentTarget}
+            />
+          </Grid>
+          <Grid item xs={8}>
+            {user && (
+              <Stack>
```
Comment (djhi) [inline mobile/src/board/components/BoardView.tsx:96]: The Stack is probably useless here. I didn't see you had a more complicated layout and the Grid alone is enough, right?

**Comment by jonathan-marmelab** [comment]
> Can you add a screenshot?

Fait!

**Review by djhi — APPROVED** [review]

File: mobile/src/board/components/BoardView.tsx (hunk 7 above)
Comment (jonathan-marmelab) [reply mobile/src/board/components/BoardView.tsx]: The stack is effectively useless, I removed it

**Review by djhi — APPROVED** [review]

**Pull request merged**